source/features/highest-rated-comment.tsx

```diff
@@ -4,48 +4,78 @@ import select from 'select-dom';
 import features from '../libs/features';
 import * as icons from '../libs/icons';
 
-function getCount(reaction: HTMLElement): number {
-	return Number(/\d+/.exec(reaction.textContent!)![0]);
-}
+// `.js-timeline-item` gets the nearest comment excluding the very first comment (OP post)
+const COMMENT_SELECTOR = '.js-timeline-item';
+
+const positiveReactions = [
+	'[aria-label*="reacted with thumbs up"]',
+	'[aria-label*="reacted with hooray"]',
+	'[aria-label*="reacted with heart"]'
+];
+
+const negativeReactions = [
+	'[aria-label*="reacted with thumbs down"]'
+];
 
 function init(): false | void {
+	const bestComment = getBestComment();
+	if (!bestComment) {
+		return false;
+	}
+
+	highlightBestComment(bestComment);
+	linkBestComment(bestComment);
+}
+
+function getBestComment(): HTMLElement | null {
 	let highest;
-	// `.js-timeline-item` excludes the very first comment
-	for (const like of select.all('.js-timeline-item [aria-label*="reacted with thumbs up"]')) {
-		const count = getCount(like);
-		const dislike = select('[aria-label*="reacted with thumbs down"]', like.parentElement!);
+	for (const comment of getCommentsWithReactions()) {
+		const likes = getPositiveReactions(comment);
+		const dislikes = getNegativeReactions(comment);
+
+		const likeCount = getCount(likes);
+		const dislikeCount = getCount(dislikes);
 
-		if (dislike && getCount(dislike) >= count / 2) {
-			continue; // Controversial comment
+		// Controversial comment, ignore
+		if (dislikeCount >= likeCount / 2) {
+			continue;
 		}
 
-		if (!highest || count > highest.count) {
-			highest = {like, count};
+		if (!highest || likeCount > highest.count) {
+			highest = {comment, count: likeCount};
 		}
 	}
 
-	if (!highest || highest.count < 10) {
-		return false;
+	// If count is not high enough don't bother telling user
+	if (!highest || highest.count < 10 || !highest.comment) {
+		return null;
 	}
 
-	const event = highest.like.closest('.js-timeline-item')!;
-	const text = select('.comment-body', event)!.textContent!.substring(0, 100);
-	const avatar = select('.timeline-comment-avatar', event)!.cloneNode(true);
-	const {hash} = select<HTMLAnchorElement>('.timestamp', event)!;
+	return highest.comment;
+}
 
-	select('.unminimized-comment', event)!.classList.add('rgh-highest-rated-comment');
-	select('.unminimized-comment .timeline-comment-header-text', event)!.before(
+function highlightBestComment(bestComment: HTMLElement): void {
+	select('.unminimized-comment', bestComment)!.classList.add('rgh-highest-rated-comment');
+	select('.unminimized-comment .timeline-comment-header-text', bestComment)!.before(
 		<span
 			className="timeline-comment-label tooltipped tooltipped-n"
 			aria-label="This comment has the most positive reactions on this issue."
 		>
 			Highest-rated comment
 		</span>
 	);
+}
+
+function linkBestComment(bestComment: HTMLElement): void {
+	// Find position of comment in thread
+	const position = select.all('.js-timeline-item').indexOf(bestComment);
```

Comment:
```suggestion
	const position = select.all(COMMENT_SELECTOR).indexOf(bestComment);
```

There's a button to apply these suggestions in a click right here on GitHub ⬆️. It will also resolve this comment.

Reply:
Oh that's cool I never saw that before. I just did it manually anyways though thanks!